code/game/objects/items/teleportation.dm

```diff
@@ -312,3 +312,154 @@
+			do_sparks(2, 1, src)
+			qdel(src)
+
+/obj/item/teleporter/proc/attempt_teleport(mob/user, EMP_D = FALSE)
+	if(!charges)
+		to_chat(user, "<span class='warning'>[src] is still recharging.</span>")
+		return
+
+	var/turf/current_location = get_turf(user)
+	var/area/current_area = current_location.loc
+	if(!current_location || !(current_area.area_flags & NOTELEPORT) || is_away_level(current_location.z) || is_centcom_level(current_location.z) || !isturf(user.loc))//If turf was not found or they're on z level 2 or >7 which does not currently exist. or if user is not located on a turf
+		to_chat(user, "<span class='notice'>\The [src] is malfunctioning.</span>")
+		return
+
+	var/mob/living/carbon/C = user
+	var/teleport_distance = rand(minimum_teleport_distance,maximum_teleport_distance)
+	var/turf/destination = get_teleport_loc(current_location,C,teleport_distance,0,0,0,0,0,0)
+	var/list/bagholding = user.GetAllContents(/obj/item/storage/backpack/holding)
+
+	if(isclosedturf(destination))
+		if(!EMP_D && !(bagholding.len))
+			panic_teleport(user, destination) //We're in a wall, engage emergency parallel teleport.
+		else
+			get_fragged(user, destination) //EMP teleported you into a wall? Wearing a BoH? You're dead.
+	else
+		telefrag(destination, user)
+		do_teleport(C, destination, channel = TELEPORT_CHANNEL_FREE)
+		charges--
+		check_charges()
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(current_location)
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
+		playsound(destination, 'sound/effects/phasein.ogg', 25, 1)
+		playsound(destination, "sparks", 50, 1)
+
+/obj/item/teleporter/proc/panic_teleport(mob/user, turf/destination)
+	var/mob/living/carbon/C = user
+	var/turf/mobloc = get_turf(C)
+	var/turf/emergency_destination = get_teleport_loc(destination,C,0,0,1,saving_throw_distance,0,0,0)
+
+	if(emergency_destination)
+		telefrag(emergency_destination, user)
+		do_teleport(C, emergency_destination, channel = TELEPORT_CHANNEL_FREE)
+		charges--
+		check_charges()
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(emergency_destination)
+		playsound(emergency_destination, 'sound/effects/phasein.ogg', 25, 1)
+		playsound(emergency_destination, "sparks", 50, 1)
+	else //We tried to save. We failed. Death time.
+		get_fragged(user, destination)
+
+/obj/item/teleporter/proc/get_fragged(mob/user, turf/destination)
+	var/turf/mobloc = get_turf(user)
+	user.forceMove(destination)
+	playsound(mobloc, "sparks", 50, TRUE)
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
+	playsound(destination, "sparks", 50, TRUE)
+	playsound(destination, "sound/magic/disintegrate.ogg", 50, TRUE)
+	to_chat(user, "<span class='userdanger'>You teleport into the wall, the teleporter tries to save you, but-</span>")
```

Comment:
`to_chat(user, span_userdanger("You teleport into the wall, the teleporter tries to save you, but-"))`